python/lsst/ts/m2gui/controltab/tab_hardpoint_selection.py

```diff
@@ -0,0 +1,300 @@
+                self._callback_suggest_hardpoints,
+                tool_tip="Suggest the hardpoints",
+            ),
+        }
+
+        self.set_widget_and_layout()
+
+        self._set_signal_detailed_force(
+            self.model.utility_monitor.signal_detailed_force
+        )
+
+    @asyncSlot()
+    async def _callback_reset_default(self) -> None:
+        """Callback of the reset-default button to reset the selection of
+        hardpoints back to the default."""
+
+        await run_command(self._set_hardpoints, self._hardpoints)
+
+    def _set_hardpoints(self, hardpoints: list[int]) -> None:
+        """Set the hardpoints.
+
+        Parameters
+        ----------
+        hardpoints : `list`
+            List of the 0-based hardpoints. There are 6 actuators. The first
+            three are the axial actuators and the latter three are the tangent
+            links.
+        """
+
+        for idx, hardpoint in enumerate(self._buttons_hardpoint_selection):
+            hardpoint.setChecked(idx in hardpoints)
+
+    @asyncSlot()
+    async def _callback_apply_hardpoints(self) -> None:
+        """Callback of the apply-hardpoints button to apply the selected
+        hardpoints."""
+
+        await run_command(self._set_hardpoint_list)
+
+    async def _set_hardpoint_list(self) -> None:
+        """Set the hardpoint list.
+
+        Raises
+        ------
+        `ValueError`
+            When do not select 6 hardpoints.
+        `RuntimeError`
+            When not in the Diagnostic state.
+        `RuntimeError`
+            When not in the simulation mode.
+        """
+
+        # Check the selection
+        hardpoints = get_checked_buttons(self._buttons_hardpoint_selection)
+        if len(hardpoints) != 6:
+            raise ValueError("Please select 6 hardpoints.")
+
+        check_hardpoints(
+            self._cell_geom["locAct_axial"],
+            hardpoints[:NUM_HARDPOINTS_AXIAL],
+            hardpoints[NUM_HARDPOINTS_AXIAL:],
+        )
+
+        # Check the state
+        allowed_mode = LocalMode.Diagnostic
+        if self.model.local_mode != allowed_mode:
+            raise RuntimeError(f"Only allowed in {allowed_mode!r}")
+
+        # Only support in the simulation mode at this moment.
+        # TODO: Remove this after the M2 cell LabVIEW code supports this.
+        if not self.model._is_simulation_mode:
+            raise RuntimeError("Only support in the simulation mode.")
+
+        # Set the hardpoints
+        await self.model.controller.set_hardpoint_list(hardpoints)
+
+    @asyncSlot()
+    async def _callback_suggest_hardpoints(self) -> None:
+        """Callback of the suggest-hardpoints button to suggest the possible
+        hardpoints."""
+
+        await run_command(self._suggest_hardpoints)
+
+    def _suggest_hardpoints(self) -> None:
+        """Suggest the hardpoints.
+
+        Raises
+        ------
+        `ValueError`
+            When there is no selected axial hardpoint.
```

Comment:
Suggest the hardpoints. Requires at least one axial and one tangential hardpoints are already selected.

Raises
------
`ValueError`
    When there aren't at least one axial and one tangential actuators already selected.

Reply:
Improved in the commit: [7931cfb](https://github.com/lsst-ts/ts_m2gui/pull/60/commits/7931cfb20f28855c11e7a6dab714c376171f3ede). Thanks!